Makefile

```diff
@@ -3,86 +3,47 @@ export PATH := $(PATH):$(GOPATH)/bin
 INTEGRATION     := nginx
 BINARY_NAME      = nri-$(INTEGRATION)
 SRC_DIR          = ./src/
-VALIDATE_DEPS    = golang.org/x/lint/golint
-TEST_DEPS        = github.com/axw/gocov/gocov github.com/AlekSi/gocov-xml
 INTEGRATIONS_DIR = /var/db/newrelic-infra/newrelic-integrations/
 CONFIG_DIR       = /etc/newrelic-infra/integrations.d
 GO_FILES        := ./src/
+GOFLAGS			 = -mod=mod
+GOLANGCI_LINT	 = github.com/golangci/golangci-lint/cmd/golangci-lint
+GOCOV            = github.com/axw/gocov/gocov
+GOCOV_XML		 = github.com/AlekSi/gocov-xml
 
 all: build
 
 build: clean validate compile test
 
 clean:
-	@echo "=== $(INTEGRATION) === [ clean ]: removing binaries and coverage file..."
+	@echo "=== $(INTEGRATION) === [ clean ]: cleaning workspace..."
 	@rm -rfv bin coverage.xml
 
-validate-deps:
-	@echo "=== $(INTEGRATION) === [ validate-deps ]: installing validation dependencies..."
-	@go get -v $(VALIDATE_DEPS)
-
-validate-only:
+validate:
 ifeq ($(strip $(GO_FILES)),)
 	@echo "=== $(INTEGRATION) === [ validate ]: no Go files found. Skipping validation."
 else
-	@printf "=== $(INTEGRATION) === [ validate ]: running gofmt... "
-	@OUTPUT="$(shell gofmt -l $(GO_FILES))" ;\
-	if [ -z "$$OUTPUT" ]; then \
-		echo "passed." ;\
-	else \
-		echo "failed. Incorrect syntax in the following files:" ;\
-		echo "$$OUTPUT" ;\
-		exit 1 ;\
-	fi
-	@printf "=== $(INTEGRATION) === [ validate ]: running golint... "
-	@OUTPUT="$(shell golint $(SRC_DIR)...)" ;\
-	if [ -z "$$OUTPUT" ]; then \
-		echo "passed." ;\
-	else \
-		echo "failed. Issues found:" ;\
-		echo "$$OUTPUT" ;\
-		exit 1 ;\
-	fi
-	@printf "=== $(INTEGRATION) === [ validate ]: running go vet... "
-	@OUTPUT="$(shell go vet $(SRC_DIR)...)" ;\
-	if [ -z "$$OUTPUT" ]; then \
-		echo "passed." ;\
-	else \
-		echo "failed. Issues found:" ;\
-		echo "$$OUTPUT" ;\
-		exit 1;\
-	fi
+	@printf "=== $(INTEGRATION) === [ validate ]: running golangci-lint... "
+	@go run $(GOFLAGS) $(GOLANGCI_LINT) run --verbose
 endif
 
-validate: validate-deps validate-only
-
-compile-deps:
-	@echo "=== $(INTEGRATION) === [ compile-deps ]: installing build dependencies..."
-	@go get -v -d -t ./...
-
 bin/$(BINARY_NAME):
 	@echo "=== $(INTEGRATION) === [ compile ]: building $(BINARY_NAME)..."
 	@go build -v -o bin/$(BINARY_NAME) $(GO_FILES)
 
-compile: compile-deps bin/$(BINARY_NAME)
+compile: bin/$(BINARY_NAME)
 
-test-deps: compile-deps
-	@echo "=== $(INTEGRATION) === [ test-deps ]: installing testing dependencies..."
-	@go get -v $(TEST_DEPS)
-
-test-only:
+test:
 	@echo "=== $(INTEGRATION) === [ test ]: running unit tests..."
-	@gocov test ./... | gocov-xml > coverage.xml
-
-test: test-deps test-only
+	@go run $(GOFLAGS) $(GOCOV) test ./... | go run $(GOFLAGS) $(GOCOV_XML) > coverage.xml
```

Comment:
just for curiosity why is the `-mod=mod` needed? I guess is related to the new tool using approach

Reply:
yes, the go command will ignore the vendor directory and use the go.mod